File: JavaAlgorithm/src/main/java/algo/tzashinorpu/FourthRound/Chapter03/mergeTwoLists_21.java
package algo.tzashinorpu.FourthRound.Chapter03;

public class mergeTwoLists_21 {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        // 1->2->4
        // 5
        ListNode dummyNode = new ListNode(0);
        ListNode concat = dummyNode;
        while (list1 != null && list2 != null) {
            if (list1.val <= list2.val) {
                concat.next = list1;
                list1 = list1.next;
            } else {
                concat.next = list2;
                list2 = list2.next;
            }
            concat = concat.next;
        }
        if (list1 == null) {
            concat.next = list2;
        }
        if (list2 == null) {
            concat.next = list1;
        }
        return dummyNode.next;
    }
}

File: Clion/ZeroTrac/array_list/mergeTwoLists_21.cpp
//
// Created by TzashiNorpu on 10/17/2022.
//
#include "array_list.h"

using namespace Array_List;

ListNode *Solution::mergeTwoLists(ListNode *list1, ListNode *list2) {
  ListNode *dummyNode = new ListNode;
  ListNode *p = dummyNode;
  while (list1 && list2) {
    if (list1->val < list2->val)
      p->next = list1, list1 = list1->next;
    else
      p->next = list2, list2 = list2->next;
    p = p->next;
  }
  if (list1) p->next = list1;
  if (list2) p->next = list2;
  return dummyNode->next;
}